### PR TITLE
Fix #123: simpler pagination with .pages() without first .get

### LINKED ISSUE #123: Simplify pagination
Pagination currently works this way:
```
statuses  = api.statuses_user_timeline().get()
for item in statuses().pages():
    print(item.id)
```

We should be able to do it in a more straightforward and simpler way: 
```
for item in api.statuses_user_timeline().pages():
    print(item.id)
```

### PR DESCRIPTION
**Pull request by fjsj**
Closes #123

### CODE REVIEW

File: tapioca/tapioca.py [hunk 1]
```diff
@@ -261,8 +261,19 @@ def _reached_max_limits(self, page_count, item_count, max_pages,
...
+        if self._response is not None:
+            if params is not None:
+                raise Exception("Since you're paging after the first .get call, "
+                                "you can't pass params here."
+                                "Pass params on the .get call instead.")
+            executor = self
+        else:
+            # this mean .pages() was called before the first .get,
+            # like `api.statuses_user_timeline().pages()`
+            response = self.get(params=params)
```
Comment (filipeximenes) [inline tapioca/tapioca.py]: Shouldn't we simply pass all the `kwargs`?
Passing only the params will limit the options that can be passed in the request, a header for example.

File: tapioca/tapioca.py [hunk 2]
```diff
@@ -261,8 +261,19 @@ def _reached_max_limits(self, page_count, item_count, max_pages,
...
-        executor = self
+    def pages(self, max_pages=None, max_items=None, params=None, **kwargs):
+        if self._response is not None:
+            if params is not None:
+                raise Exception("Since you're paging after the first .get call, "
+                                "you can't pass params here."
+                                "Pass params on the .get call instead.")
+            executor = self
+        else:
+            # this mean .pages() was called before the first .get,
```
Comment (filipeximenes) [inline tapioca/tapioca.py]: `this means ...`

File: tapioca/tapioca.py [hunk 3]
```diff
@@ -261,8 +261,19 @@ def _reached_max_limits(self, page_count, item_count, max_pages,
         reached_item_limit = max_items is not None and max_items <= item_count
         return reached_page_limit or reached_item_limit
 
-    def pages(self, max_pages=None, max_items=None, **kwargs):
-        executor = self
+    def pages(self, max_pages=None, max_items=None, params=None, **kwargs):
+        if self._response is not None:
+            if params is not None:
+                raise Exception("Since you're paging after the first .get call, "
```
Comment (filipeximenes) [inline tapioca/tapioca.py]: Just improving the message:
`You cannot pass new params after the first request is done. Use the ".get" method to pass parameters`

**Comment by fjsj** [comment]
Closing because this would be a major change in API.

**Pull request closed without merging**